### PR TITLE
Add monitoring for DNS flakiness

### PR DESCRIPTION
**Pull request by chaodaiG**

### CODE REVIEW

File: tools/monitoring/config/config.yaml [hunk 1]
```diff
@@ -152,3 +152,17 @@ spec:
         jobs-affected: 1
         prs-affected: 0
         period: 1440
+
+  - error-pattern: 'Could not resolve host: github.com'
```
Comment (yt3liu) [inline tools/monitoring/config/config.yaml]: Move this error to the top of the file. 
@steuhs Correct me if I'm wrong, but I believe the error matching logic only scans through the config top down and only select the first config that it finds.

Comment (chaodaiG) [reply tools/monitoring/config/config.yaml]: had an in person discussion with @steuhs , this doesn't apply for this pattern

**Pull request merged**